docs/src/references/synchronize-daemons.md

```diff
@@ -0,0 +1,116 @@
+ So, better to check the status of the bridge before sending the request to the c8y cloud.
+
+3. Some of the daemons are dependent on the `tedge-mapper-c8y`(cloud mapper) file system to create the supported operations file.
+
+# Proposed solutions and their Pros and Cons
+
+## Using the `health check` mechanism to synchronize `tedge-mapper-c8y` and `tedge-agent`
+
+- The `tedge-mapper-c8y` can use the health-check mechanism to detect the health of the `tedge-agent` and then send the request to the `tedge-agent`.
+- When the mapper starts it will subscribe to the `tedge/health/tedge-agent` topic.
+- When the `tedge-agent` starts, it will publish a `health` status (up/down) message on the `tedge/health/tedge-agent` topic.
+- Based on this `tedge-agent` health status message, the mapper can create the supported operations in the `/etc/tedge/operations/c8y` directory.
+- Check if the `c8y-bridge` is up before sending the init messages, including `supported operations` list.
+- Mapper will send the supported operations list to the c8y cloud.
+- Mapper will send the request `software list` to `tedge-agent`.
+- Once it receives a software list from `tedge-agent`, the mapper will send the list to the c8y cloud.
+
+### Communicating with Cumulocity over HTTP
+
+The `tedge-mapper-c8y` needs `internal-id` of the `thin-edge` device from Cumulocity to communicate over HTTP.
+When the tedge-mapper-c8y starts, it will request the `internal id` of the `thin-edge` device from cumulocity cloud.
+Which will be used by the mapper for further communication with the c8y cloud over HTTP.
+
+To get the `internal id` over HTTP, the mapper needs the `JWT` token, which was retrieved over the `MQTT`.
+So, before sending the request to retrieve the `JWT` token on the `c8y/s/uat` topic,
+the mapper must check the health of the `c8y bridge`.
+When the bridge is up, it can send a request for the JWT token.
+Then the mapper will receive the `JWT` token on the `c8y/s/dat` topic.
+
+If there is any issue in retrieving the JWT token then the mapper must `re-try` again.
+
+Once the mapper possesses the JWT token, it can get the thin-edge device's internal-id from Cumulocity.
+If there is any issue with getting the internal-id then the mapper must `re-try` getting the
+internal id.
+
+Once the `tedge-mapper-c8y` possesses the `internal-id`, the mapper can start using HTTP to communicate with the Cumulocity.
+While communicating with Cumulocity, If there is any issue it must check the reason for the error and,
+If the failure is due to the `internal-id` then it must get the internal id again and re-try communication with Cumulocify over HTTP.
+
+For example, The mapper sends software update/list operation result to the c8y cloud over `HTTP`.
+If there is any issue sending the operation status and list, it must check the reason for the error and,
+If the failure is due to the `internal-id` then it must get the internal id again and must re-try sending the operation result.
+
+
```

Comment:
Even though caching of the `jwt` token and `internal-id` are not mentioned in this spec, I hope your impl would be doing that anyway. In-memory caching to start with. We can build the persistance bits later on.